src/main/java/com/kodedu/boot/AppStarter.java

```diff
@@ -270,6 +272,19 @@ private void registerStartupListener(CmdlineConfig config) {
                     }
                 });
             });
+            threadService.runActionLater(() -> {
```

Comment:
@Ayowel 
It is a nice idea to define working directory dynamically. What do you think if we make .asciidocfx as a hidden folder? Later we can put more config files under such folder and AsciidocFX can read some configs from that folder ?

I think here not a specific number like 12, the logic can traverse from current path to root volume. Root directories' parent will be null, so it can stop if parent is null or a .asciidocfx folder is found. Another stop condition can be if process has not access permission to the resolved parent.

What do you think ?

Reply:
I wonder if changing the working directory should be done at Tab change/add, as adding this feature as-is would mean opening/closing AsciidocFX each time a workdir change is required (or updating it manually), which makes this feature less intuitive. What is your take on this ?

    What do you think if we make .asciidocfx as a hidden folder? Later we can put more config files under such folder and AsciidocFX can read some configs from that folder ?

Sounds good to me. I think what could be done is just check for the existence of a file or directory for now, so that an empty file can be created while no configuration values are expected and support for a specific subdirectory layout could be added when the need arises. If we only support a directory, it would be good to define a default expected filename so as to not pollute versioning tools (ie git) by forcing projects using them to create a file with a specific name.

    I think here not a specific number like 12, the logic can traverse from current path to root volume.
    
I wanted a sanity check as I wasn't sure how getParent behaves on non-windows filesystem (I know that `/`'s parent is itself, so wanted to avoid a case where a thread would be stuck in an infinite loop by adding this validation. I can remove it if you don't think it's relevant.

    Another stop condition can be if process has not access permission to the resolved parent.

Would you consider sufficient a check for the reachability of the target file ? Permissions can be set-up in weird ways and I'd rather keep validation as stupid as possible.

Reply:
@Ayowel Sorry, for the late return, I pushed one commit about the proposed changes https://github.com/asciidocfx/AsciidocFX/commit/d8ce46180d2d37923cb971c178b8f9044b93526e Could you please review it ?

Reply:
Looks good to me. Added comments because i like to nitpick.